internal/hcsoci/layers.go

```diff
@@ -142,7 +142,7 @@ func MountContainerLayers(ctx context.Context, layerFolders []string, guestRoot
 
 	hostPath := filepath.Join(layerFolders[len(layerFolders)-1], "sandbox.vhdx")
 	containerScratchPathInUVM := ospath.Join(uvm.OS(), guestRoot)
-	_, _, err = uvm.AddSCSI(ctx, hostPath, containerScratchPathInUVM, false)
+	_, _, containerScratchPathInUVM, err = uvm.AddSCSI(ctx, hostPath, containerScratchPathInUVM, false)
```

Comment:
We have talked about this at length in the past. We really should be returning a type here. And then you can call "release" on that type rather than even having a remove. Would be much cleaner impl

Reply:
But that is a large change unfortunately

Reply:
@abwah - I'm concerned that we will fail somewhere on the teardown paths if we dont move `containerScratchPathInUVM` to a `global` location under `/run/mounts`. If we let a container own its mounts then shared mounts cant be destroyed until all containers are done. But with @katiewasnothere recent code we now delete the `/run/containers/<id>` folder on exit. So we would either fail or delete an active mount.

Reply:
@jterry75 I have moved the VHD mounts to /run/mounts which are ref counted. Please have a look